### ISSUE TITLE
Lambda writeReplace function error.

### ISSUE THREAD

**Issue by siaron** [quarkiverse/quarkus-mybatis#184]
com.baomidou.mybatisplus.core.toolkit.LambdaUtils
```
/**
     * 该缓存可能会在任意不定的时间被清除
     *
     * @param func 需要解析的 lambda 对象
     * @param <T>  类型，被调用的 Function 对象的目标类型
     * @return 返回解析后的结果
     */
    public static <T> LambdaMeta extract(SFunction<T, ?> func) {
        // 1. IDEA 调试模式下 lambda 表达式是一个代理
        if (func instanceof Proxy) {
            return new IdeaProxyLambdaMeta((Proxy) func);
        }
        // 2. 反射读取
        try {
            Method method = func.getClass().getDeclaredMethod("writeReplace");
            return new ReflectLambdaMeta((SerializedLambda) ReflectionKit.setAccessible(method).invoke(func));
        } catch (Throwable e) {
            // 3. 反射失败使用序列化的方式读取
            return new ShadowLambdaMeta(com.baomidou.mybatisplus.core.toolkit.support.SerializedLambda.extract(func));
        }
    }
```
在使用queryWrapper的时候这里报错没? 这里好像会报错spring native 和 mp 集成时.

**Comment by zhfeng** [comment]
Can you share any re-producer codes ?

**Comment by siaron** [comment]
demo
[quarkus-mp.zip](https://github.com/quarkiverse/quarkus-mybatis/files/8407085/quarkus-mp.zip)
1.  maven Compile to native: `./mvnw package -Pnative`
2. run : `./quarkus-mp-1.0-SNAPSHOT-runner`
3. http get `http://localhost:8080/mybatis/plus/user/search` 
4.  http method:
```
    @Path("/user/search")
    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public List<User> search() {
        LambdaQueryWrapper<User> userQueryWrapper = Wrappers.<User>lambdaQuery()
                .eq(User::getId, 1);
        return userMapper.selectList(userQueryWrapper);
    }
```

**Comment by zhfeng** [comment]
Thanks and it seems that mybatis-plus wants to resolve a class from the serialize lambda func. I'm not very this could be supportted in native mode. I'll keep investigating.

**Comment by zhfeng** [comment]
Well, the lambda (class) bytecodes are generated at runtime. So there is no way to register the reflections when quarkus builds the application. But it looks like graalvm will support for lambda class serialization in Native Image, see https://github.com/oracle/graal/issues/3756 and there is  a commit https://github.com/oracle/graal/commit/b455f23b624a1af149c7d5769443a4775bedd087 which I think it tried to resolve the issue. Unforunally this commit has not been included in any graalvm release.

So maybe we can wait for a new graalvm release in the future to support lambda class serialization.

**Comment by zhfeng** [comment]
@siaron I just tried to build the latest graalvm locally. and yeah, it supports the lambda class serialization when I add
```
{
   "types":[],
   "lambdaCapturingTypes":[
              {"name":"com.quarkus.demo.mp.MyBatisPlusResource"}
           ]
}
```
in the serialization-config.json. So let's wait for the new release of graalvm.

@gastaldi is it possible to support this new feature ```lambdaCapturingTypes``` in quarkus?

**Comment by gastaldi** [comment]
@zhfeng probably yes once it is available in the GraalVM recommended to Quarkus. Can you open an issue in the Quarkus issue tracker requesting this?

**Comment by siaron** [comment]
> Well, the lambda (class) bytecodes are generated at runtime. So there is no way to register the reflections when quarkus builds the application. But it looks like graalvm will support for lambda class serialization in Native Image, see [oracle/graal#3756](https://github.com/oracle/graal/issues/3756) and there is a commit [oracle/graal@b455f23](https://github.com/oracle/graal/commit/b455f23b624a1af149c7d5769443a4775bedd087) which I think it tried to resolve the issue. Unforunally this commit has not been included in any graalvm release.
> 
> So maybe we can wait for a new graalvm release in the future to support lambda class serialization.

yes . I have also seen this

**Comment by zhfeng** [comment]
@gastaldi I raise https://github.com/quarkusio/quarkus/issues/24861 to track it.

**Comment by siaron** [comment]
any update？

**Comment by zhfeng** [comment]
It looks like the fix was include in graalvm 22.1.0 and quarkus has been upgraded. So I think it should be good to take a look to https://github.com/quarkusio/quarkus/issues/24861

**Comment by siaron** [comment]
quarkus [2.10.0.CR1](https://github.com/quarkusio/quarkus/releases/tag/2.10.0.CR1) support it?

**Comment by zhfeng** [comment]
@siaron yeah, I think we had fixed https://github.com/quarkusio/quarkus/issues/24861 in [2.10.0.CR1](https://github.com/quarkusio/quarkus/releases/tag/2.10.0.CR1)

**Comment by siaron** [comment]
how to use? Is it currently not possible to use it directly?

**Comment by zhfeng** [comment]
I think you can upgrade `quarkus.version` to `2.10.0.CR1` at first.

And then you can try to build `quarkus-mybatis` locally with `1.0.4-SNAPSHOT`. Use the following anntation

```java
@RegisterForReflection(lambdaCapturingTypes = "com.quarkus.demo.mp.MyBatisPlusResource", 
                       targets = { SerializedLambda.class, SFunction.class }, 
                       serialization = true)
@Path("/mybatis/plus")
public class MyBatisPlusResource {

    @Path("/user/search")
    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public List<User> search() throws Exception {
      LambdaQueryWrapper<User> userQueryWrapper = Wrappers.<User>lambdaQuery()
                .eq(User::getId, 1);
        return userMapper.selectList(userQueryWrapper);
    }

}
```

Can you verify it in native mode by running with `mvn clean verify -Dnative`?

**Comment by siaron** [comment]
> I think you can upgrade `quarkus.version` to `2.10.0.CR1` at first.
> 
> And then you can try to build `quarkus-mybatis` locally with `1.0.4-SNAPSHOT`. Use the following anntation
> 
> ```java
> @RegisterForReflection(lambdaCapturingTypes = "com.quarkus.demo.mp.MyBatisPlusResource", 
>                        targets = { SerializedLambda.class, SFunction.class }, 
>                        serialization = true)
> @Path("/mybatis/plus")
> public class MyBatisPlusResource {
> 
>     @Path("/user/search")
>     @GET
>     @Produces(MediaType.APPLICATION_JSON)
>     public List<User> search() throws Exception {
>       LambdaQueryWrapper<User> userQueryWrapper = Wrappers.<User>lambdaQuery()
>                 .eq(User::getId, 1);
>         return userMapper.selectList(userQueryWrapper);
>     }
> 
> }
> ```
> 
> Can you verify it in native mode by running with `mvn clean verify -Dnative`?

very nice, it work normal.

**Comment by zhfeng** [comment]
Hi @siaron 

I'm going to do a new release with bumping to quarkus 2.10..Final. So you can try  `quarkus-mybatis:1.0.4` later.

**Comment by zhfeng** [comment]
It should be fixed in quarkus `2.10.0.Final`
